pkg/controllers/nodeclaim/lifecycle/liveness.go

```diff
@@ -51,6 +55,30 @@ func (l *Liveness) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (reco
 	if ttl := registrationTTL - l.clock.Since(registered.LastTransitionTime.Time); ttl > 0 {
 		return reconcile.Result{RequeueAfter: ttl}, nil
 	}
+	nodePool := &v1.NodePool{}
+	if err := l.kubeClient.Get(ctx, types.NamespacedName{Name: nodeClaim.Labels[v1.NodePoolLabelKey]}, nodePool); err != nil {
+		return reconcile.Result{}, err
+	}
+	if nodePool.StatusConditions().Get(v1.ConditionTypeNodeRegistrationHealthy).IsUnknown() {
+		stored := nodePool.DeepCopy()
+		// If the nodeClaim failed to launch/register during the TTL set NodeRegistrationHealthy status condition on
+		// NodePool to False. If the launch failed get the launch failure reason and message from nodeClaim.
+		if nodeClaim.StatusConditions().IsTrue(v1.ConditionTypeLaunched) {
```

Comment:
What do you think about getting this once and storing it in the if statement? Then it can be used in the else to get the launch failure.
```suggestion
		if launchCondition := nodeClaim.StatusConditions().Get(v1.ConditionTypeLaunched); launchCondition.IsTrue() {
```